helpers/traits/AltapayMaster.php

```diff
@@ -48,38 +48,53 @@ public function scheduledSubscriptionsPayment( $amount, $renewal_order ) {
 				return;
 			}
 
-			$parent_order_id = 0;
-			if ( wcs_order_contains_renewal( $renewal_order->id ) ) {
-				$parent_order_id = WC_Subscriptions_Renewal_Order::get_parent_order_id( $renewal_order->id );
+			if ( wcs_order_contains_renewal( $renewal_order->get_id() ) ) {
+				$subscriptions = wcs_get_subscriptions_for_renewal_order( $renewal_order->get_id() );
 			}
 
-			$orderinfo      = new WC_Order( $parent_order_id );
-			$transaction_id = $orderinfo->get_transaction_id();
-
-			if ( ! $transaction_id ) {
-				// Set subscription payment as failure
-				$renewal_order->update_status( 'failed', __( 'AltaPay could not locate transaction ID', 'altapay' ) );
-				return;
-			}
-
-			$login = $this->altapayApiLogin();
-			if ( ! $login || is_wp_error( $login ) ) {
-				echo '<p><b>' . __( 'Could not connect to AltaPay!', 'altapay' ) . '</b></p>';
-				return;
-			}
-
-			$api = new ChargeSubscription( $this->getAuth() );
-			$api->setTransaction( $transaction_id );
-			$api->setAmount( round( $amount, 2 ) );
-			$response = $api->call();
-
-			if ( $response->Result === 'Success' ) {
-				$renewal_order->payment_complete();
-			} else {
-				$renewal_order->update_status(
-					'failed',
-					sprintf( __( 'AltaPay payment declined: %s', 'altapay' ), $response->MerchantErrorMessage )
-				);
+			foreach ( $subscriptions as $subscription ) {
+				$parent_order_id = $subscription->get_parent_id();
+				$agreement_id    = get_post_meta( $parent_order_id, '_agreement_id', true );
+
+				if ( ! $agreement_id ) {
+					// Set subscription payment as failure
+					$renewal_order->update_status( 'failed', __( 'AltaPay could not locate agreement ID', 'altapay' ) );
+					return;
+				}
+
+				$login = $this->altapayApiLogin();
+				if ( ! $login || is_wp_error( $login ) ) {
+					echo '<p><b>' . __( 'Could not connect to AltaPay!', 'altapay' ) . '</b></p>';
+					return;
+				}
+
+				// @phpstan-ignore-next-line
+				if ( $this->payment_action === 'authorize_capture' ) {
+					$api = new ChargeSubscription( $this->getAuth() );
+					$api->setTransaction( $agreement_id );
```

Comment:
Ins't there an agreementId to be used in the SDK instead?

https://documentation.altapay.com/Content/Ecom/Scenarios/Charge%20Agreement.htm

Reply:
@emicha We have added `setAgreement()` method as well in the SDK but `transaction_id` is defined as mandatory parameter. 
So, in the next SDK release, we need to make `transaction_id` parameter optional for specific payment operations.

Reply:
@shahbaz-mehar That is right, and then allow the merchant to define the `agreement[id]` for subscriptions, instead of `transaction_id`.